chevah/compat/tests/normal/test_filesystem.py

```diff
@@ -1715,6 +1715,95 @@ def test_openFileForReading_no_write(self):
             if a_file:
                 a_file.close()
 
+    def test_openFileForReading_no_delete_lock(self):
+        """
+        A file opened only for reading will not be locked for delete
+        operations.
+        """
+        _, segments = mk.fs.makePathInTemp()
+        mk.fs.writeFileContent(segments=segments, content='something-\N{sun}')
+        a_file = None
+        try:
+            a_file = self.filesystem.openFileForReading(segments)
+
+            self.filesystem.deleteFile(segments)
+
+            content = a_file.read(100)
+            self.assertEqual(b'something-\xe2\x98\x89', content)
+        finally:
+            if a_file:
+                a_file.close()
+
+    def test_openFile_read_no_delete_lock(self):
+        """
+        A file opened only for reading will not be locked for delete
+        operations.
+
+        The file is opened in binary mode.
+        """
+        content = 'first-line\nmore-lines\r\nsomething-\N{sun}'
+        _, segments = mk.fs.makePathInTemp()
+        mk.fs.writeFileContent(segments=segments, content=content)
+        fd = None
+        try:
+            fd = self.filesystem.openFile(
+                segments,
+                flags=self.filesystem.OPEN_READ_ONLY,
+                mode=0,
+                )
+
+            self.filesystem.deleteFile(segments)
```

Comment:
Maybe also check that the file WAS deleted (such as with `os.stat`).

Reply:
I think this was ok.
If `filesystem.deleteFile` is not doing what it should, we should have a separate test and fix for the `delteFile` method.

To be extra safe, I went with:

```diff
            self.filesystem.deleteFile(segments)
+           self.assertFalse(self.filesystem.exists(segments))
```